filler option not working correctly

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When using the `filler` option in a right-aligned block, my prompt does not render correctly.

Here is a screenshot *without* filler:
![nofiller](https://github.com/JanDeDobbeleer/oh-my-posh/assets/959529/b3baed2a-2483-4db1-b818-4798e2ac12d9)

Here is a screenshot *with* filler:
![filler](https://github.com/JanDeDobbeleer/oh-my-posh/assets/959529/369b08c1-38d0-4610-8a88-d3e74fa5d2d3)

Notice how the right side of prompt does not look right. It appears as if my filler is one character too wide

The exact line I added was:

```
"filler": "<#ffffff,transparent>-"
```

I made sure to add it to a *right aligned* block.

I'm using oh my posh running in Powershell 7 in Windows Terminal on Windows 11.



### Theme

I'm using my own theme which is a variation on powerlevel10k_rainbow

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "version": 2,
  "final_space": true,
  "console_title_template": "{{ .Shell }} in {{ .Folder }}",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "os",
          "style": "diamond",
          "foreground": "#000000",
          "background": "#d3d7cf",
          "leading_diamond": "\u256d\u2500\ue0b2",
          "template": " {{ if .WSL }}WSL at {{ end }}{{.Icon}} "
        },
        {
          "type": "path",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#e4e4e4",
          "background": "#3465a4",
          "template": " \uf07c {{ path .Path .Location }} ",
          "properties": {
            "home_icon": "~",
            "style": "full"
          }
        },
        {
          "type": "git",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#000000",
          "background": "#4e9a06",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#c4a000{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#f26d50{{ end }}",
            "{{ if gt .Ahead 0 }}#89d1dc{{ end }}",
            "{{ if gt .Behind 0 }}#4e9a06{{ end }}"
          ],
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{ .BranchStatus }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \uf692 {{ .StashCount }}{{ end }} ",
          "properties": {
            "branch_icon": "\uf126 ",
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          }
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "right",
      "filler": "<#ffffff,transparent>-",
      "segments": [
        {
          "type": "node",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#689f63",
          "template": " {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }} \uf898 ",
          "properties": {
            "fetch_version": true
          }
        },
        {
          "type": "go",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#111111",
          "background": "#00acd7",
          "template": " {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} \ue627 ",
          "properties": {
            "fetch_version": true
          }
        },
        {
          "type": "julia",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#111111",
          "background": "#4063D8",
          "template": " {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} \ue624 ",
          "properties": {
            "fetch_version": true
          }
        },
        {
          "type": "python",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#111111",
          "background": "#FFDE57",
          "template": " {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} \ue235 ",
          "properties": {
            "display_mode": "files",
            "fetch_version": true,
            "fetch_virtual_env": false
          }
        },
        {
          "type": "ruby",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#AE1401",
          "template": " {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} \ue791 ",
          "properties": {
            "display_mode": "files",
            "fetch_version": true
          }
        },
        {
          "type": "azfunc",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#FEAC19",
          "template": " {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} \uf0e7",
          "properties": {
            "display_mode": "files",
            "fetch_version": false
          }
        },
        {
          "type": "aws",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background_templates": [
            "{{if contains \"default\" .Profile}}#FFA400{{end}}",
            "{{if contains \"jan\" .Profile}}#f1184c{{end}}"
          ],
          "template": " {{ .Profile }}{{ if .Region }}@{{ .Region }}{{ end }} \ue7ad ",
          "properties": {
            "display_default": false
          }
        },
        {
          "type": "root",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#111111",
          "background": "#ffff66",
          "template": " \uf0e7 ",
          "properties": {
            "root_icon": "\uf0ad"
          }
        },
        {
          "type": "executiontime",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#000000",
          "background": "#c4a000",
          "template": " {{ .FormattedMs }} \uf252 "
        },
        {
          "type": "exit",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#d3d7cf",
          "background": "#000000",
          "background_templates": [
            "{{ if gt .Code 0 }}#cc2222{{ end }}"
          ],
          "template": " {{ if gt .Code 0 }}{{ .Meaning }}{{ else }}\u2714{{ end }} ",
          "properties": {
            "always_enabled": true
          }
        },
        {
          "type": "time",
          "style": "diamond",
          "invert_powerline": true,
          "foreground": "#000000",
          "background": "#d3d7cf",
          "trailing_diamond": "\ue0b0\u2500\u256e",
          "template": " {{ .CurrentDate | date .Format }} \uf017 "
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "text",
          "style": "plain",
          "foreground": "#d3d7cf",
          "template": "\u2570\u2500"
        }
      ],
      "newline": true
    },
    {
      "type": "rprompt",
      "segments": [
        {
          "type": "text",
          "style": "plain",
          "foreground": "#d3d7cf",
          "template": "\u2500\u256f"
        }
      ]
    }
  ]
}

```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 18.2.0

Shell: pwsh (7.3.6)

Segments:

ConsoleTitle(true)                         -   0 ms - pwsh in chris
os(true)                                   -   0 ms - ╭─ 
path(true)                                 -   1 ms -   ~
git(false)                                 -   1 ms -
node(false)                                -   0 ms -
go(false)                                  -   0 ms -
julia(false)                               -   0 ms -
python(false)                              -   0 ms -
ruby(false)                                -   0 ms -
azfunc(false)                              -   0 ms -
aws(false)                                 -   0 ms -
root(false)                                -   0 ms -
executiontime(false)                       -   0 ms -
exit(true)                                 -   0 ms -  ✔
time(true)                                 -   0 ms -  13:38:33 
text(true)                                 -   0 ms - ╰─
text(true)                                 -   0 ms - ─╯

Run duration: 9.4155ms

Cache path: C:\Users\chris\AppData\Local\oh-my-posh

Config path: C:\Users\chris\Documents\clients\christof\_setup\christof.omp.json

Logs:

[DEBUG] 13:38:33.144 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.144 shell.go:Getenv(TERM_PROGRAM) - 0s
[DEBUG] 13:38:33.144 shell.go:Getenv:398 → C:\Users\chris\AppData\Local
[TRACE] 13:38:33.144 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 13:38:33.145 shell_windows.go:CachePath() - 510.5µs
[DEBUG] 13:38:33.145 shell.go:Getenv:398 → C:\Users\chris\Documents\clients\christof\_setup\christof.omp.json
[TRACE] 13:38:33.145 shell.go:Getenv(POSH_THEME) - 0s
[DEBUG] 13:38:33.145 shell.go:Shell:638 → no shell name provided in flags, trying to detect it
[DEBUG] 13:38:33.149 shell.go:Shell:646 → process name: pwsh.exe
[TRACE] 13:38:33.149 shell.go:Shell() - 3.5733ms
[TRACE] 13:38:33.149 shell.go:resolveConfigPath() - 3.5733ms
[TRACE] 13:38:33.149 shell.go:Init() - 4.1747ms
[TRACE] 13:38:33.149 shell.go:Flags() - 0s
[TRACE] 13:38:33.149 config.go:loadConfig() - 509.8µs
[TRACE] 13:38:33.149 shell.go:Flags() - 0s
[DEBUG] 13:38:33.149 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.149 shell.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 13:38:33.149 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.149 shell_windows.go:WindowsRegistryKeyValue:206 → ColorizationColor(DWORD): 0xC40078D4
[TRACE] 13:38:33.149 shell.go:Shell() - 0s
[DEBUG] 13:38:33.149 shell.go:Getenv:398 → 7.3.6
[TRACE] 13:38:33.149 shell.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 13:38:33.149 debug.go:PrintDebug:23 → Segment: Title
[TRACE] 13:38:33.149 shell_windows.go:Root() - 0s
[TRACE] 13:38:33.149 shell.go:Shell() - 0s
[TRACE] 13:38:33.149 shell.go:StatusCodes() - 0s
[TRACE] 13:38:33.149 shell_windows.go:IsWsl() - 0s
[DEBUG] 13:38:33.149 shell.go:TemplateCache:816 → environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\chris\AppData\Roaming CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=VYBE3080 ComSpec=C:\windows\system32\cmd.exe CONDA_PROMPT_MODIFIER=False configsetroot=C:\windows\ConfigSetRoot DriverData=C:\Windows\System32\Drivers\DriverData FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer FPS_BROWSER_USER_PROFILE_STRING=Default HOMEDRIVE=C: HOMEPATH=\Users\chris LOCALAPPDATA=C:\Users\chris\AppData\Local LOGONSERVER=\\VYBE3080 NUMBER_OF_PROCESSORS=20 OneDrive=C:\Users\chris\OneDrive OS=Windows_NT Path=C:\Program Files\PowerShell\7;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\PowerShell\7\;C:\Program Files\Git\cmd;C:\Program Files\nodejs\;C:\Users\chris\AppData\Local\Microsoft\WindowsApps;C:\Users\chris\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\chris\.dotnet\tools;C:\Users\chris\AppData\Local\Programs\oh-my-posh\bin;C:\Users\chris\AppData\Roaming\npm PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL POSH_AZURE_ENABLED=False POSH_CURSOR_COLUMN=1 POSH_CURSOR_LINE=33 POSH_GIT_ENABLED=False POSH_INSTALLER=winget POSH_PID=20364 POSH_SHELL_VERSION=7.3.6 POSH_THEME=C:\Users\chris\Documents\clients\christof\_setup\christof.omp.json POSH_THEMES_PATH=C:\Users\chris\AppData\Local\Programs\oh-my-posh\themes POWERLINE_COMMAND=oh-my-posh POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Pro POWERSHELL_UPDATECHECK=LTS PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 151 Stepping 2, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=9702 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=C:\Users\chris\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\windows\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SystemDrive=C: SystemRoot=C:\windows TEMP=C:\Users\chris\AppData\Local\Temp TMP=C:\Users\chris\AppData\Local\Temp USERDOMAIN=VYBE3080 USERDOMAIN_ROAMINGPROFILE=VYBE3080 USERNAME=chris USERPROFILE=C:\Users\chris windir=C:\windows WSLENV=WT_SESSION::WT_PROFILE_ID WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=7d761bde-9bba-4b8d-8ea8-e96669c0e019 ZES_ENABLE_SYSMAN=1]
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.149 shell.go:Pwd:428 → C:\Users\chris
[TRACE] 13:38:33.149 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.149 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.149 shell.go:User:549 → chris
[TRACE] 13:38:33.149 shell.go:User() - 0s
[DEBUG] 13:38:33.149 shell.go:Host:561 → VYBE3080
[TRACE] 13:38:33.149 shell.go:Host() - 0s
[TRACE] 13:38:33.149 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.149 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.149 shell.go:Getenv(SHLVL) - 0s
[TRACE] 13:38:33.149 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.150 shell.go:Flags() - 0s
[DEBUG] 13:38:33.150 block.go:Debug:280 → Segment: os
[TRACE] 13:38:33.150 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.150 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.150 properties.go:GetString:28 → 
[TRACE] 13:38:33.150 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.150 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.150 shell.go:Shell() - 0s
[DEBUG] 13:38:33.150 block.go:Debug:280 → Segment: path
[TRACE] 13:38:33.150 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.150 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:Pwd() - 0s
[TRACE] 13:38:33.150 shell.go:Shell() - 0s
[TRACE] 13:38:33.150 shell.go:Flags() - 0s
[DEBUG] 13:38:33.150 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 13:38:33.150 properties.go:GetString:28 → 
[DEBUG] 13:38:33.150 properties.go:GetString:28 → 
[DEBUG] 13:38:33.150 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.150 properties.go:GetString:28 → ~
[DEBUG] 13:38:33.150 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 13:38:33.150 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.150 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.150 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.150 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:PathSeparator() - 0s
[TRACE] 13:38:33.151 shell.go:Pwd() - 0s
[TRACE] 13:38:33.151 shell_windows.go:IsWsl() - 0s
[TRACE] 13:38:33.151 shell.go:StackCount() - 0s
[DEBUG] 13:38:33.151 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 13:38:33.151 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 13:38:33.151 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 13:38:33.151 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 13:38:33.151 shell_windows.go:DirIsWritable() - 523.9µs
[TRACE] 13:38:33.151 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.151 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.151 shell.go:Shell() - 0s
[DEBUG] 13:38:33.151 block.go:Debug:280 → Segment: git
[TRACE] 13:38:33.151 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.151 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[TRACE] 13:38:33.151 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 shell.go:CommandPath:600 → C:\Program Files\Git\cmd\git.exe
[TRACE] 13:38:33.153 shell.go:CommandPath(git.exe) - 1.5548ms
[TRACE] 13:38:33.153 shell.go:HasCommand(git.exe) - 1.5548ms
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[ERROR] 13:38:33.153 shell.go:HasParentFilePath:734 → CreateFile .git: The system cannot find the file specified.
[TRACE] 13:38:33.153 shell.go:HasParentFilePath(.git) - 0s
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → fetch_bare_info: false
[TRACE] 13:38:33.153 shell.go:Flags() - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: node
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: go
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: julia
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [*.jl]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: python
[DEBUG] 13:38:33.153 properties.go:GetString:28 → files
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv env venv-win pyenv-win]
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: ruby
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [*.rb Rakefile Gemfile]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: azfunc
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → extensions: [host.json local.settings.json function.json]
[DEBUG] 13:38:33.153 properties.go:GetStringArray:52 → folders: []
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → home_enabled: false
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: aws
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → display_default: false
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_VAULT) - 0s
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_DEFAULT_PROFILE) - 0s
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_PROFILE) - 0s
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_REGION) - 0s
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_DEFAULT_REGION) - 0s
[DEBUG] 13:38:33.153 shell.go:Getenv:398 → NO DATA
[TRACE] 13:38:33.153 shell.go:Getenv(AWS_CONFIG_FILE) - 507.5µs
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[ERROR] 13:38:33.153 shell.go:FileContent:519 → open C:\Users\chris/.aws/config: The system cannot find the path specified.
[TRACE] 13:38:33.153 shell.go:FileContent(C:\Users\chris/.aws/config) - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: root
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[TRACE] 13:38:33.153 shell_windows.go:Root() - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: executiontime
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → always_enabled: false
[TRACE] 13:38:33.153 shell.go:ExecutionTime() - 0s
[DEBUG] 13:38:33.153 properties.go:GetFloat64:34 → threshold: 500.000000
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: exit
[DEBUG] 13:38:33.153 properties.go:GetString:28 → {{ .Code }}
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[TRACE] 13:38:33.153 shell.go:StatusCodes() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[DEBUG] 13:38:33.153 properties.go:GetBool:22 → always_enabled: true
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:Shell() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: time
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[DEBUG] 13:38:33.153 properties.go:GetString:28 → 15:04:05
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:Shell() - 0s
[TRACE] 13:38:33.153 shell.go:Flags() - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: text
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:Shell() - 0s
[TRACE] 13:38:33.153 shell.go:Flags() - 0s
[DEBUG] 13:38:33.153 block.go:Debug:280 → Segment: text
[TRACE] 13:38:33.153 shell.go:Pwd() - 0s
[DEBUG] 13:38:33.153 shell_windows.go:56 → C:\Users\chris
[TRACE] 13:38:33.153 shell.go:GOOS() - 0s
[TRACE] 13:38:33.153 shell.go:TemplateCache() - 0s
[TRACE] 13:38:33.153 shell.go:Shell() - 0s
[DEBUG] 13:38:33.154 shell.go:Getenv:398 → C:\Users\chris\AppData\Local
[TRACE] 13:38:33.154 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 13:38:33.154 shell_windows.go:CachePath() - 0s
[TRACE] 13:38:33.154 shell.go:Flags() - 0s
```

--- a/src/ansi/ansi_writer.go
+++ b/src/ansi/ansi_writer.go
@@ -97,7 +97,6 @@ type Writer struct {
 	shell                 string
 	format                string
 	left                  string
-	right                 string
 	title                 string
 	linechange            string
 	clearBelow            string
@@ -127,7 +126,6 @@ func (w *Writer) Init(shellName string) {
 	case shell.BASH:
 		w.format = "\\[%s\\]"
 		w.linechange = "\\[\x1b[%d%s\\]"
-		w.right = "\\[\x1b[%dC\\]"
 		w.left = "\\[\x1b[%dD\\]"
 		w.clearBelow = "\\[\x1b[0J\\]"
 		w.clearLine = "\\[\x1b[K\\]"
@@ -144,7 +142,6 @@ func (w *Writer) Init(shellName string) {
 	case shell.ZSH, shell.TCSH:
 		w.format = "%%{%s%%}"
 		w.linechange = "%%{\x1b[%d%s%%}"
-		w.right = "%%{\x1b[%dC%%}"
 		w.left = "%%{\x1b[%dD%%}"
 		w.clearBelow = "%{\x1b[0J%}"
 		w.clearLine = "%{\x1b[K%}"
@@ -160,7 +157,6 @@ func (w *Writer) Init(shellName string) {
 		w.osc51 = "%%{\x1b]51;A%s@%s:%s\x1b\\%%}"
 	default:
 		w.linechange = "\x1b[%d%s"
-		w.right = "\x1b[%dC"
 		w.left = "\x1b[%dD"
 		w.clearBelow = "\x1b[0J"
 		w.clearLine = "\x1b[K"
@@ -193,15 +189,6 @@ func (w *Writer) SetParentColors(background, foreground string) {
 		Background: background,
 		Foreground: foreground,
 	}}, w.ParentColors...)
-}
-
-func (w *Writer) CarriageForward() string {
-	return fmt.Sprintf(w.right, 1000)
-}
-
-func (w *Writer) GetCursorForRightWrite(length, offset int) string {
-	strippedLen := length + (-offset)
-	return fmt.Sprintf(w.left, strippedLen)
 }
 
 func (w *Writer) ChangeLine(numberOfLines int) string {

--- a/src/cli/print.go
+++ b/src/cli/print.go
@@ -24,6 +24,7 @@ var (
 	shellVersion string
 	plain        bool
 	noStatus     bool
+	column       int
 )
 
 // printCmd represents the prompt command
@@ -64,6 +65,7 @@ var printCmd = &cobra.Command{
 			Primary:       args[0] == "primary",
 			Cleared:       cleared,
 			NoExitCode:    noStatus,
+			Column:        column,
 		}
 
 		eng := engine.New(flags)
@@ -107,6 +109,7 @@ func init() { //nolint:gochecknoinits
 	printCmd.Flags().BoolVarP(&plain, "plain", "p", false, "plain text output (no ANSI)")
 	printCmd.Flags().BoolVar(&cleared, "cleared", false, "do we have a clear terminal or not")
 	printCmd.Flags().BoolVar(&eval, "eval", false, "output the prompt for eval")
+	printCmd.Flags().IntVar(&column, "column", 0, "the column position of the cursor")
 	// Deprecated flags
 	printCmd.Flags().IntVarP(&status, "error", "e", 0, "last exit code")
 	printCmd.Flags().BoolVar(&noStatus, "no-exit-code", false, "no valid exit code (cancelled or no command yet)")

--- a/src/engine/engine.go
+++ b/src/engine/engine.go
@@ -35,27 +35,48 @@ func (e *Engine) string() string {
 	return text
 }
 
-func (e *Engine) canWriteRightBlock(rprompt bool) bool {
-	if rprompt && (e.rprompt == "" || e.Plain) {
-		return false
+func (e *Engine) canWriteRightBlock(rprompt bool) (int, bool) {
+	if rprompt && (len(e.rprompt) == 0 || e.Plain) {
+		return 0, false
 	}
+
 	consoleWidth, err := e.Env.TerminalWidth()
 	if err != nil || consoleWidth == 0 {
-		return true
+		return 0, false
 	}
+
 	promptWidth := e.currentLineLength
 	availableSpace := consoleWidth - promptWidth
+
 	// spanning multiple lines
 	if availableSpace < 0 {
 		overflow := promptWidth % consoleWidth
 		availableSpace = consoleWidth - overflow
 	}
+
+	if rprompt {
+		availableSpace -= e.rpromptLength
+	}
+
 	promptBreathingRoom := 5
 	if rprompt {
 		promptBreathingRoom = 30
 	}
-	canWrite := (availableSpace - e.rpromptLength) >= promptBreathingRoom
-	return canWrite
+
+	canWrite := availableSpace >= promptBreathingRoom
+
+	return availableSpace, canWrite
+}
+
+func (e *Engine) writeRPrompt() {
+	space, OK := e.canWriteRightBlock(true)
+	if !OK {
+		return
+	}
+	e.write(e.Writer.SaveCursorPosition())
+	e.write(strings.Repeat(" ", space))
+	e.write(e.rprompt)
+	e.write(e.Writer.RestoreCursorPosition())
 }
 
 func (e *Engine) pwd() {
@@ -105,23 +126,28 @@ func (e *Engine) isWarp() bool {
 	return e.Env.Getenv("TERM_PROGRAM") == "WarpTerminal"
 }
 
-func (e *Engine) shouldFill(filler string, length int) (string, bool) {
+func (e *Engine) shouldFill(filler string, blockLength int) (string, bool) {
 	if len(filler) == 0 {
 		return "", false
 	}
+
 	terminalWidth, err := e.Env.TerminalWidth()
 	if err != nil || terminalWidth == 0 {
 		return "", false
 	}
-	padLength := terminalWidth - e.currentLineLength - length
+
+	padLength := terminalWidth - e.currentLineLength - blockLength
 	if padLength <= 0 {
 		return "", false
 	}
+
+	// allow for easy color overrides and templates
 	e.Writer.Write("", "", filler)
 	filler, lenFiller := e.Writer.String()
 	if lenFiller == 0 {
 		return "", false
 	}
+
 	repeat := padLength / lenFiller
 	return strings.Repeat(filler, repeat), true
 }
@@ -197,7 +223,7 @@ func (e *Engine) renderBlock(block *Block, cancelNewline bool) {
 		text, length := block.RenderSegments()
 		e.rpromptLength = length
 
-		if !e.canWriteRightBlock(false) {
+		if _, OK := e.canWriteRightBlock(false); OK {
 			switch block.Overflow {
 			case Break:
 				e.newline()
@@ -216,6 +242,7 @@ func (e *Engine) renderBlock(block *Block, cancelNewline bool) {
 			e.write(text)
 			return
 		}
+
 		prompt := e.Writer.CarriageForward()
 		prompt += e.Writer.GetCursorForRightWrite(length, block.HorizontalOffset)
 		prompt += text

--- a/src/engine/engine.go
+++ b/src/engine/engine.go
@@ -126,17 +126,12 @@ func (e *Engine) isWarp() bool {
 	return e.Env.Getenv("TERM_PROGRAM") == "WarpTerminal"
 }
 
-func (e *Engine) shouldFill(filler string, blockLength int) (string, bool) {
+func (e *Engine) shouldFill(filler string, remaining, blockLength int) (string, bool) {
 	if len(filler) == 0 {
 		return "", false
 	}
 
-	terminalWidth, err := e.Env.TerminalWidth()
-	if err != nil || terminalWidth == 0 {
-		return "", false
-	}
-
-	padLength := terminalWidth - e.currentLineLength - blockLength
+	padLength := remaining - blockLength
 	if padLength <= 0 {
 		return "", false
 	}
@@ -221,32 +216,36 @@ func (e *Engine) renderBlock(block *Block, cancelNewline bool) {
 		}
 
 		text, length := block.RenderSegments()
-		e.rpromptLength = length
 
-		if _, OK := e.canWriteRightBlock(false); OK {
+		space, OK := e.canWriteRightBlock(false)
+		// we can't print the right block as there's not enough room available
+		if !OK {
 			switch block.Overflow {
 			case Break:
 				e.newline()
 			case Hide:
 				// make sure to fill if needed
-				if padText, OK := e.shouldFill(block.Filler, 0); OK {
+				if padText, OK := e.shouldFill(block.Filler, space, 0); OK {
 					e.write(padText)
 				}
+				e.currentLineLength = 0
 				return
 			}
 		}
 
-		if padText, OK := e.shouldFill(block.Filler, length); OK {
-			// in this case we can print plain
+		defer func() {
+			e.currentLineLength = 0
+		}()
+
+		// validate if we have a filler and fill if needed
+		if padText, OK := e.shouldFill(block.Filler, space, length); OK {
 			e.write(padText)
 			e.write(text)
 			return
 		}
 
-		prompt := e.Writer.CarriageForward()
-		prompt += e.Writer.GetCursorForRightWrite(length, block.HorizontalOffset)
+		prompt := strings.Repeat(" ", space-length)
 		prompt += text
-		e.currentLineLength = 0
 		e.write(prompt)
 	case RPrompt:
 		e.rprompt, e.rpromptLength = block.RenderSegments()

--- a/src/engine/engine_test.go
+++ b/src/engine/engine_test.go
@@ -21,7 +21,7 @@ func TestCanWriteRPrompt(t *testing.T) {
 		PromptLength       int
 		RPromptLength      int
 	}{
-		{Case: "Width Error", Expected: true, TerminalWidthError: errors.New("burp")},
+		{Case: "Width Error", Expected: false, TerminalWidthError: errors.New("burp")},
 		{Case: "Terminal > Prompt enabled", Expected: true, TerminalWidth: 200, PromptLength: 100, RPromptLength: 10},
 		{Case: "Terminal > Prompt enabled edge", Expected: true, TerminalWidth: 200, PromptLength: 100, RPromptLength: 70},
 		{Case: "Prompt > Terminal enabled", Expected: true, TerminalWidth: 200, PromptLength: 300, RPromptLength: 70},
@@ -39,7 +39,7 @@ func TestCanWriteRPrompt(t *testing.T) {
 			currentLineLength: tc.PromptLength,
 			rprompt:           "hello",
 		}
-		got := engine.canWriteRightBlock(true)
+		_, got := engine.canWriteRightBlock(true)
 		assert.Equal(t, tc.Expected, got, tc.Case)
 	}
 }

--- a/src/engine/prompt.go
+++ b/src/engine/prompt.go
@@ -44,6 +44,7 @@ func (e *Engine) Primary() string {
 
 	if e.Config.FinalSpace {
 		e.write(" ")
+		e.currentLineLength++
 	}
 
 	if e.Config.ShellIntegration && e.Config.TransientPrompt == nil {
@@ -59,11 +60,7 @@ func (e *Engine) Primary() string {
 		}
 		// Warp doesn't support RPROMPT so we need to write it manually
 		if e.isWarp() {
-			e.write(e.Writer.SaveCursorPosition())
-			e.write(e.Writer.CarriageForward())
-			e.write(e.Writer.GetCursorForRightWrite(e.rpromptLength, 0))
-			e.write(e.rprompt)
-			e.write(e.Writer.RestoreCursorPosition())
+			e.writeRPrompt()
 			// escape double quotes contained in the prompt
 			prompt := fmt.Sprintf("PS1=\"%s\"", strings.ReplaceAll(e.string(), `"`, `\"`))
 			return prompt
@@ -73,16 +70,10 @@ func (e *Engine) Primary() string {
 		prompt += fmt.Sprintf("\nRPROMPT=\"%s\"", e.rprompt)
 		return prompt
 	case shell.PWSH, shell.PWSH5, shell.GENERIC, shell.NU:
-		if !e.canWriteRightBlock(true) {
-			break
-		}
-		e.write(e.Writer.SaveCursorPosition())
-		e.write(e.Writer.CarriageForward())
-		e.write(e.Writer.GetCursorForRightWrite(e.rpromptLength, 0))
-		e.write(e.rprompt)
-		e.write(e.Writer.RestoreCursorPosition())
+		e.writeRPrompt()
 	case shell.BASH:
-		if !e.canWriteRightBlock(true) {
+		space, OK := e.canWriteRightBlock(true)
+		if !OK {
 			break
 		}
 		// in bash, the entire rprompt needs to be escaped for the prompt to be interpreted correctly
@@ -92,8 +83,7 @@ func (e *Engine) Primary() string {
 		}
 		writer.Init(shell.GENERIC)
 		prompt := writer.SaveCursorPosition()
-		prompt += writer.CarriageForward()
-		prompt += writer.GetCursorForRightWrite(e.rpromptLength, 0)
+		prompt += strings.Repeat(" ", space)
 		prompt += e.rprompt
 		prompt += writer.RestoreCursorPosition()
 		prompt = e.Writer.FormatText(prompt)

--- a/src/engine/prompt.go
+++ b/src/engine/prompt.go
@@ -152,8 +152,11 @@ func (e *Engine) ExtraPrompt(promptType ExtraPromptType) string {
 
 	str, length := e.Writer.String()
 	if promptType == Transient {
-		if padText, OK := e.shouldFill(prompt.Filler, length); OK {
-			str += padText
+		consoleWidth, err := e.Env.TerminalWidth()
+		if err == nil || consoleWidth != 0 {
+			if padText, OK := e.shouldFill(prompt.Filler, consoleWidth, length); OK {
+				str += padText
+			}
 		}
 	}
 

--- a/src/engine/tooltip.go
+++ b/src/engine/tooltip.go
@@ -49,12 +49,22 @@ func (e *Engine) Tooltip(tip string) string {
 		if !block.Enabled() {
 			return ""
 		}
+
+		consoleWidth, err := e.Env.TerminalWidth()
+		if err != nil || consoleWidth == 0 {
+			return ""
+		}
+
 		text, length := block.RenderSegments()
+
+		space := consoleWidth - e.Env.Flags().Column - length
+		if space <= 0 {
+			return ""
+		}
 		// clear from cursor to the end of the line in case a previous tooltip
 		// is cut off and partially preserved, if the new one is shorter
 		e.write(e.Writer.ClearAfter())
-		e.write(e.Writer.CarriageForward())
-		e.write(e.Writer.GetCursorForRightWrite(length, 0))
+		e.write(strings.Repeat(" ", space))
 		e.write(text)
 		return e.string()
 	}

--- a/src/platform/shell.go
+++ b/src/platform/shell.go
@@ -73,6 +73,7 @@ type Flags struct {
 	Cleared       bool
 	TrueColor     bool
 	NoExitCode    bool
+	Column        int
 }
 
 type CommandError struct {

--- a/src/platform/shell_unix.go
+++ b/src/platform/shell_unix.go
@@ -55,15 +55,20 @@ func (env *Shell) IsWsl2() bool {
 
 func (env *Shell) TerminalWidth() (int, error) {
 	defer env.Trace(time.Now())
+
 	if env.CmdFlags.TerminalWidth > 0 {
 		env.DebugF("terminal width: %d", env.CmdFlags.TerminalWidth)
 		return env.CmdFlags.TerminalWidth, nil
 	}
+
 	width, err := terminal.Width()
 	if err != nil {
 		env.Error(err)
 	}
-	return int(width), err
+
+	env.DebugF("terminal width: %d", width)
+	env.CmdFlags.TerminalWidth = int(width)
+	return env.CmdFlags.TerminalWidth, err
 }
 
 func (env *Shell) Platform() string {

--- a/src/platform/shell_unix.go
+++ b/src/platform/shell_unix.go
@@ -55,7 +55,8 @@ func (env *Shell) IsWsl2() bool {
 
 func (env *Shell) TerminalWidth() (int, error) {
 	defer env.Trace(time.Now())
-	if env.CmdFlags.TerminalWidth != 0 {
+	if env.CmdFlags.TerminalWidth > 0 {
+		env.DebugF("terminal width: %d", env.CmdFlags.TerminalWidth)
 		return env.CmdFlags.TerminalWidth, nil
 	}
 	width, err := terminal.Width()

--- a/src/platform/shell_windows.go
+++ b/src/platform/shell_windows.go
@@ -87,7 +87,8 @@ func (env *Shell) IsWsl2() bool {
 
 func (env *Shell) TerminalWidth() (int, error) {
 	defer env.Trace(time.Now())
-	if env.CmdFlags.TerminalWidth != 0 {
+	if env.CmdFlags.TerminalWidth > 0 {
+		env.DebugF("terminal width: %d", env.CmdFlags.TerminalWidth)
 		return env.CmdFlags.TerminalWidth, nil
 	}
 	handle, err := syscall.Open("CONOUT$", syscall.O_RDWR, 0)

--- a/src/platform/shell_windows.go
+++ b/src/platform/shell_windows.go
@@ -87,22 +87,27 @@ func (env *Shell) IsWsl2() bool {
 
 func (env *Shell) TerminalWidth() (int, error) {
 	defer env.Trace(time.Now())
+
 	if env.CmdFlags.TerminalWidth > 0 {
 		env.DebugF("terminal width: %d", env.CmdFlags.TerminalWidth)
 		return env.CmdFlags.TerminalWidth, nil
 	}
+
 	handle, err := syscall.Open("CONOUT$", syscall.O_RDWR, 0)
 	if err != nil {
 		env.Error(err)
 		return 0, err
 	}
+
 	info, err := winterm.GetConsoleScreenBufferInfo(uintptr(handle))
 	if err != nil {
 		env.Error(err)
 		return 0, err
 	}
-	// return int(float64(info.Size.X) * 0.57), nil
-	return int(info.Size.X), nil
+
+	env.CmdFlags.TerminalWidth = int(info.Size.X)
+	env.DebugF("terminal width: %d", env.CmdFlags.TerminalWidth)
+	return env.CmdFlags.TerminalWidth, nil
 }
 
 func (env *Shell) Platform() string {

--- a/src/shell/scripts/omp.ps1
+++ b/src/shell/scripts/omp.ps1
@@ -131,8 +131,9 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
                 return
             }
             $position = $host.UI.RawUI.CursorPosition
+            $terminalWidth = $Host.UI.RawUI.WindowSize.Width
             $cleanPSWD = Get-CleanPSWD
-            $standardOut = @(Start-Utf8Process $script:OMPExecutable @("print", "tooltip", "--status=$script:ErrorCode", "--shell=$script:ShellName", "--pswd=$cleanPSWD", "--config=$env:POSH_THEME", "--command=$command", "--shell-version=$script:PSVersion"))
+            $standardOut = @(Start-Utf8Process $script:OMPExecutable @("print", "tooltip", "--status=$script:ErrorCode", "--shell=$script:ShellName", "--pswd=$cleanPSWD", "--config=$env:POSH_THEME", "--command=$command", "--shell-version=$script:PSVersion", "--column=$($position.X)", "--terminal-width=$terminalWidth"))
             # ignore an empty tooltip
             if ($standardOut -eq '') {
                 return

--- a/src/shell/scripts/omp.ps1
+++ b/src/shell/scripts/omp.ps1
@@ -417,6 +417,13 @@ Example:
             $terminalWidth = 0
         }
 
+        # in some cases we have an empty $script:NoExitCode
+        # this is a workaround to make sure we always have a value
+        # see https://github.com/JanDeDobbeleer/oh-my-posh/issues/4128
+        if ($null -eq $script:NoExitCode) {
+            $script:NoExitCode = $true
+        }
+
         # set the cursor positions, they are zero based so align with other platforms
         $env:POSH_CURSOR_LINE = $Host.UI.RawUI.CursorPosition.Y + 1
         $env:POSH_CURSOR_COLUMN = $Host.UI.RawUI.CursorPosition.X + 1

